R/AllGenerics.R

```diff
@@ -22,6 +22,9 @@ setGeneric("deleteSpectra", function(object, ...)
 setGeneric("IonDb", function(x, cdb, ...)
     standardGeneric("IonDb"))
 
+setGeneric("mass2mz", function(x, adduct = c("[M+H]+"))
```

Comment:
I would suggest to change here the `adduct = c("[M+H]+")` to simply `adduct` - I would not define default values in the generic method definition.

Reply:
Also, to support additional parameters I would add `, ...` to the generic method definition (same as for `IonDb` above).

Reply:
I've added the `...` and removed the `adduct` parameter altogether from the generic, since, when it was missing in the function call, it would not use the method's default "[M+H]+":

> Error in .local(x, adduct, ...) : 
  argument "adduct" is missing, with no default

An alternative would be keeping the adduct parameter in the generic and creating a new method for signature c("CompDb", "missing") that sets the default value and redirects to the c("CompDb", "character") method. But I think that'd be more complex than needed, right?

Reply:
hm, then maybe we go for your solution to add `...` and drop `adduct` in the generic definition. That sounds best to me.